script.js

```diff
@@ -1,4 +1,943 @@
+
+  return gameLogicDisplay + message2;
+};
+
+//Game Logic for Multiplayer Mode
+var arrangeDiceMultiplayer = function (input) {
+  leaderBoardArray = [];
+
+  //control for highest or lowest mode toggle
+  if (
+    input.toLowerCase() == "" ||
+    input.toLowerCase() == "highest" ||
+    input.toLowerCase() == "lowest"
+  ) {
+    gameCount = gameCount + 1;
+
+    //GAME LOGIC AND GAME LOGIC CONTROL
+    if (
+      (gameLogic == "highest" && input.toLowerCase() == "highest") ||
+      (gameLogic == "highest" && input.toLowerCase() == "") ||
+      (gameLogic == "lowest" && input.toLowerCase() == "highest")
+    ) {
+      gameLogic = "highest";
+      gameLogicDisplay = `Game Logic: ${gameLogic.toUpperCase()} Combined Number<br><br>`;
+    } else if (
+      (gameLogic == "lowest" && input.toLowerCase() == "lowest") ||
+      (gameLogic == "lowest" && input.toLowerCase() == "") ||
+      (gameLogic == "highest" && input.toLowerCase() == "lowest")
+    ) {
+      gameLogic = "lowest";
+      gameLogicDisplay = `Game Logic: ${gameLogic.toUpperCase()} Combined Number<br><br>`;
+    }
+
+    // Local Variables for arrangeDice Function.
+    // To Generate the Combined number based on putting together N number dice rolls.
+
+    var diceRollHistory = [];
+    var diceRollsArraySorted = [];
+    var diceRollsArraySortedString = [];
+    var combinedNumberString = "";
+    var combinedNumber = 0;
+    var incumbentIndex = "";
+    var incumbentCombinedNumber = 0;
+    var winnerThisRound = "";
+    var leaderBoardArray = [];
+    var roundHistoryDisplay = [];
+    var playerAisplay = ""; // just using a "previous winner" or incumbent holder isn't enough, the playerA- playerB needs a whole assignment
+    var playerBdisplay = ""; // on its own.
+
+    // Loop to run through number of players and get their rolls automatically
+    while (playerIndexNmbr < numberOfPlayers) {
+      var diceRollsArray = [];
+      var count = 0;
+
+      diceNumberRolled = 0;
+
+      while (count < qtyOfDice) {
+        diceNumberRolled = diceRoll();
+        diceRollsArray.push(diceNumberRolled);
+        diceRollHistory.push(diceNumberRolled);
+        count = count + 1;
+
+        //sorting the diceRoll Array
+        if (gameLogic == "highest") {
+          diceRollsArraySorted = diceRollsArray.sort((a, b) => b - a);
+          diceRollsArraySortedString = diceRollsArraySorted.map(String);
+        } else if (gameLogic == "lowest") {
+          diceRollsArraySorted = diceRollsArray.sort((a, b) => a - b);
+          diceRollsArraySortedString = diceRollsArraySorted.map(String);
+        }
+      }
+      //combining the diceRoll array to form number
+      combinedNumberString = diceRollsArraySortedString.join("");
+      combinedNumber = Number(combinedNumberString);
+
+      //Current Player Assignment + their scores
+      playerName = "Player " + (playerIndexNmbr + 1);
```

Comment:
Having this variable being named playerName isn't really accurate as it is more accurate to call it playerID